mne/time_frequency/csd.py

```diff
@@ -1,63 +1,522 @@
-# Author: Roman Goj <roman.goj@gmail.com>
+# -*- coding: utf-8 -*-
+# Authors: Marijn van Vliet <w.m.vanvliet@gmail.com>
+#          Susanna Aro <susanna.aro@aalto.fi>
+#          Roman Goj <roman.goj@gmail.com>
 #
 # License: BSD (3-clause)
 
 import copy as cp
+import numbers
 
 import numpy as np
-
-from ..io.pick import pick_types
-from ..utils import logger, verbose, warn
+from .tfr import cwt, morlet
+from ..io.pick import pick_types, pick_channels
+from ..utils import logger, verbose, warn, copy_function_doc_to_method_doc
+from ..viz.misc import plot_csd
 from ..time_frequency.multitaper import (dpss_windows, _mt_spectra,
                                          _csd_from_mt, _psd_from_mt_adaptive)
-
-from ..externals.six.moves import xrange as range
+from ..parallel import parallel_func
+from ..externals.h5io import read_hdf5, write_hdf5
 
 
 class CrossSpectralDensity(object):
     """Cross-spectral density.
 
+    Given a list of time series, the CSD matrix denotes for each pair of time
+    series, the cross-spectral density. This matrix is symmetric and internally
+    stored as a vector.
+
+    This object can store multiple CSD matrices: one for each frequency.
+    Use ``.get_matrix(freq)`` to obtain an CSD matrix as an ndarray.
```

Comment:
get_matrix to get an array is confusing. About get_data?

Reply:
I like that.